src/main/java/com/aws/greengrass/secretmanager/SecretManager.java

```diff
@@ -333,26 +343,49 @@ public String validateSecretId(String secretId) throws GetSecretException {
         if (Utils.isEmpty(secretId)) {
             throw new GetSecretException(400, "SecretId absent in the request");
         }
-        try {
-            return getArnFromCache(secretId);
-        } catch (GetSecretException e) {
+        String arn = getArnFromCache(secretId);
+        if (!isSecretPresentInCache(arn)) {
             try {
                 reloadCache();
-                return getArnFromCache(secretId); // throw GetSecretException if the secret doesn't exist
-            } catch (SecretManagerException ex) {
-                throw new GetSecretException(404, secretNotFoundErr + secretId);
+            } catch (SecretManagerException e) {
+                // TODO: Improve return code where device is offline and SM is unable to load cache from disk
+                logger.atWarn().setCause(e).log("Unable to load secrets from cache");
             }
         }
+        return arn;
     }
 
-    private String getArnFromCache(String secretId) throws GetSecretException {
+    private boolean isSecretIdArn(final String secretId) {
+        return Pattern.matches(VALID_SECRET_ARN_PATTERN, secretId);
+    }
+
+    private boolean doesSecretNameMatchWithConfiguredArn(String secretName, String secretArn) {
+        String regex = ".*:secret:" + Pattern.quote(secretName) + "-[A-Za-z0-9]{6}(?::[A-Za-z0-9/_+=.@-]+)?$";
+        Pattern pattern = Pattern.compile(regex);
+        return pattern.matcher(secretArn).matches();
+    }
+
+    // TODO: add validation for labels
+    private String getArnFromCache(final String secretId) throws GetSecretException {
         String arn = secretId;
-        boolean isNotAnArn = !Pattern.matches(VALID_SECRET_ARN_PATTERN, secretId);
-        if (isNotAnArn) {
+        if (!isSecretIdArn(secretId)) {
             arn = nameToArnMap.get(secretId);
         }
-        if (arn == null || !cache.containsKey(arn)) {
-            throw new GetSecretException(404, secretNotFoundErr + secretId);
+
+        /*
+        In case device was offline or security service was unavailable during the initial sync, nameToArnMap
+        is empty and if GetSecret IPC is invoked using a secret name with refresh, we fail.
+        Fix: try to lookup the secret ARN using the requested secret name in configured secrets.
+         */
+        if (Utils.isEmpty(arn)) {
+            List<SecretConfiguration> configurations = getSecretConfiguration();
+            Optional<SecretConfiguration> matchedSecretConfig = configurations.stream()
+                    .filter(config -> doesSecretNameMatchWithConfiguredArn(secretId, config.getArn()))
```

Comment:
We want to do this configuration validation check with not only with the secret name but also by the label passed in the ipc request.

Reply:
same as above